bin/bench-prover/src/utils.rs

```diff
@@ -0,0 +1,86 @@
+use std::{
+    env,
+    fs::{self},
+    io::Write,
+    path::{Path, PathBuf},
+    process::Command,
+};
+
+use serde::Deserialize;
+use serde_json::{Result as JsonResult, Value, json};
+
+pub fn cargo_target_directory() -> Option<PathBuf> {
+    #[derive(Deserialize)]
+    struct Metadata {
+        target_directory: PathBuf,
+    }
+
+    env::var_os("CARGO_TARGET_DIR").map(PathBuf::from).or_else(|| {
+        let output = Command::new(env::var_os("CARGO")?)
+            .args(["metadata", "--format-version", "1"])
+            .output()
+            .ok()?;
+        let metadata: Metadata = serde_json::from_slice(&output.stdout).ok()?;
+        Some(metadata.target_directory)
+    })
+}
+
+pub fn process_benchmark_data(benchmark_path: &Path) -> JsonResult<Value> {
+    let mut benchmark_data = json!({});
+
+    // Process benchmark.json
+    let benchmark_content =
+        fs::read_to_string(benchmark_path.join("benchmark.json")).unwrap_or_default();
+    if !benchmark_content.is_empty() {
+        let json: Value = serde_json::from_str(&benchmark_content).unwrap_or_default();
+        benchmark_data["id"] = json["full_id"].clone();
+    }
+
+    // Process estimates.json
+    let estimates_content =
+        fs::read_to_string(benchmark_path.join("estimates.json")).unwrap_or_default();
+    if !estimates_content.is_empty() {
+        let json: Value = serde_json::from_str(&estimates_content).unwrap_or_default();
+
+        // Extract metrics directly with unwrap
+        let mean = json["mean"]["point_estimate"].as_f64().unwrap_or(0.0);
+        benchmark_data["mean_sec"] = json!(mean / 1_000_000_000.0);
+
+        let lower = json["mean"]["confidence_interval"]["lower_bound"].as_f64().unwrap_or(0.0);
+        benchmark_data["mean_lower_bound_sec"] = json!(lower / 1_000_000_000.0);
+
+        let upper = json["mean"]["confidence_interval"]["upper_bound"].as_f64().unwrap_or(0.0);
+        benchmark_data["mean_upper_bound_sec"] = json!(upper / 1_000_000_000.0);
+
+        let std_dev = json["std_dev"]["point_estimate"].as_f64().unwrap_or(0.0);
+        benchmark_data["std_dev_sec"] = json!(std_dev / 1_000_000_000.0);
+    }
```

Comment:
I think I would prefer not silently defaulting to `0.0`. I think it would be confusing to find `0` in the final benchmark data. I would rather error if that happens (e.g. using `anyhow`) or `expect`ing, if those values should always be `f64`s (which looks like is the case?).

This applies to basically all kinds of `unwrap`s in this file.